flashinfer/quantization/kernels/nvfp4_quantize.py

```
@@ -2001,6 +2055,68 @@ def nvfp4_quantize_cute_dsl(
+        raise ValueError("input must be a 2-D CUDA BF16 tensor")
+    m, k = input.shape
+    if k % NVFP4_SF_VEC_SIZE != 0:
+        raise ValueError(f"K ({k}) must be divisible by {NVFP4_SF_VEC_SIZE}")
+    if (
+        pre_quant_scale.dtype != torch.bfloat16
+        or not pre_quant_scale.is_cuda
+        or pre_quant_scale.numel() != k
+    ):
+        raise ValueError("pre_quant_scale must be a CUDA BF16 tensor with K elements")
+
+    input = input.contiguous()
```

Ruff (0.16.1)

[error] Variable `input` is shadowing a Python builtin

(A001)